app/helpers/flash_messages_helper.rb

```diff
@@ -40,9 +40,8 @@ def render_flash_messages
       .reject { |k, _| k.to_s == "op_primer_flash" }
       .reject { |k, _| k.start_with? "_" }
       .map do |k, v|
-      if k.to_sym == :modal
-        component = v[:type].constantize
-        component.new(**v.fetch(:parameters, {})).render_in(self)
+      if k.to_sym == :op_modal
+        render_op_modal_flash_component(component: v[:component], parameters: v.fetch(:parameters, {}))
```

Comment:
Couldn't this be simply? 

```suggestion
        render_op_modal_flash_component(**v)
```